### PR TITLE
fix: meaningful subject for mailto link

### PR DESCRIPTION
**Pull request by DorisWongNHS**
Adding meaningful subject to the "mailto" link.

**Programme notifications:**
eg. Normal subject text - GMC: 1234567, Prog No: ABC1234

**Placement notifications:**
eg. Normal subject text - GMC: 1234567, Start Date: 00/00/0000

**If the data is missing, it will appear as "unknown":**
eg. Normal subject text - GMC: unknown, Start Date: unknown

TIS21-5893
TIS21-5930
TIS21-5931
TIS21-5932

### CODE REVIEW

File: src/test/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipServiceTest.java [hunk 1]
```diff
@@ -283,7 +297,13 @@ void shouldAddInAppNotificationsWhenNotExcludedAndMeetsCriteria(NotificationType
     Map<String, Object> variables = variablesCaptor.getValue();
     assertThat("Unexpected programme name.", variables.get(PROGRAMME_NAME_FIELD),
         is(PROGRAMME_NAME));
+    assertThat("Unexpected programme number.", variables.get(PROGRAMME_NUMBER_FIELD),
+        is(PROGRAMME_NUMBER));
     assertThat("Unexpected start date.", variables.get(START_DATE_FIELD), is(START_DATE));
+    if (notificationType.equals(LTFT) || notificationType.equals(DEFERRAL)
+        || notificationType.equals(SPONSORSHIP)) {
```
Comment (ReubenRobertsHEE) [inline src/test/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipServiceTest.java:304]: Is there a reason not to check E_PORTFOLIO or INDEMNITY_INSURANCE here?

**Review by ReubenRobertsHEE — CHANGES_REQUESTED** [review]
We should include a few tests in the `EmailServiceIntegrationTest` suite to demonstrate the correct formatting of the new `mailto:` links in the templates I think (one when GMC="" and one where it is a non-empty string, same for Prog No). What do you think?

File: src/test/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipServiceTest.java (hunk 1 above)
Comment (DorisWongNHS) [reply src/test/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipServiceTest.java:304]: As the E_PORTFOLIO and INDEMNITY_INSURANCE do not have any mailto link, so the gmcNumber is not passed in there in the [createInAppNotifications()](https://github.com/Health-Education-England/tis-trainee-notifications/blob/7d8340bfbd159f4b8674c9c225e2e426c16c5b6c/src/main/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipService.java#L265)

**Review by ReubenRobertsHEE — APPROVED** [review]
Looks great! 👍

**Pull request merged**